### PR TITLE
Complete the implementation of frequency_encoder.R and the respective tests for it

### LINKED ISSUE #25: Code frequency_encoder
This is just an issue to track the progress of the entire function (since the milestone instructions state to create an issue for each function, and we only created issues for each docstring in particular).

- [x] Docstring
- [x]  Code

Will modify as we understand what we need to do in the coming weeks.

### LINKED ISSUE #31: Code test_frequency_encoder
This is just an issue to track the progress of the entire unit test function for  #24.

- [x] Docstring
- [x] 2 to 3 edge case tests

### PR DESCRIPTION
**Pull request by bbaillie**
#25 #31

### CODE REVIEW

File: R/frequency_encoder.R [hunk 1]
```diff
@@ -2,13 +2,13 @@
 get_encoding <-  function(data, col){
   out <- data %>%
     dplyr::group_by(!!sym(col))%>%
```
Comment (braydentang1) [inline R/frequency_encoder.R]:
```suggestion
    dplyr::group_by(!!rlang::sym(col))%>%
```

File: R/frequency_encoder.R [hunk 2]
```diff
@@ -2,13 +2,13 @@
...
-    dplyr::summarise(freq = n())
+    dplyr::summarise(freq = n()/nrow(data))
 }
 
 encode <- function(data,encoding_col,col){
   encoded_col <- data %>%
     dplyr::left_join(encoding_col) %>%
-    dplyr::mutate(!!sym(col) := if_else(is.na(freq), 0L, freq) # if encoding didn't exist encode value as 0
+    dplyr::mutate(!!sym(col) := if_else(is.na(freq), 0, freq) # if encoding didn't exist encode value as 0
```
Comment (braydentang1) [inline R/frequency_encoder.R:11]:
```suggestion
    dplyr::mutate(!!rlang::sym(col) := dplyr::if_else(is.na(freq), 0, freq) # if encoding didn't exist encode value as 0
```

**Review by braydentang1 — CHANGES_REQUESTED** [review]

File: R/frequency_encoder.R (hunk 1 above)
Comment (bbaillie) [reply R/frequency_encoder.R]: good catch

**Review by braydentang1 — APPROVED** [review]
